arangod/VocBase/Properties/PlanCollection.h

```diff
@@ -0,0 +1,194 @@
+
+#include <string>
+
+namespace arangodb {
+template<typename T>
+class ResultT;
+
+namespace inspection {
+struct Status;
+}
+namespace velocypack {
+class Slice;
+}
+
+struct PlanCollection {
+  PlanCollection();
+
+  static ResultT<PlanCollection> fromCreateAPIBody(
+      arangodb::velocypack::Slice input);
+
+  // Temporary method to handOver information from
+  arangodb::velocypack::Builder toCollectionsCreate();
+
+  std::string name;
+  std::underlying_type_t<TRI_col_type_e> type;
+  bool waitForSync;
+  bool isSystem;
+  bool doCompact;
+  bool isVolatile;
+  bool cacheEnabled;
+
+  uint64_t numberOfShards;
+  uint64_t replicationFactor;
+  uint64_t writeConcern;
+  std::string distributeShardsLike;
+  std::string smartJoinAttribute;
+  std::string shardingStrategy;
+  std::string globallyUniqueId;
+
+  std::vector<std::string> shardKeys;
+
+  // TODO: This can be optimized into it's own struct.
+  // Did a short_cut here to avoid concatenated changes
+  arangodb::velocypack::Builder computedValues;
+
+  // TODO: This can be optimized into it's own struct.
+  // Did a short_cut here to avoid concatenated changes
+  arangodb::velocypack::Builder schema;
+
+  // TODO: This can be optimized into it's own struct.
+  // Did a short_cut here to avoid concatenated changes
+  arangodb::velocypack::Builder keyOptions;
+
+  // NOTE: This attributes are not documented
+  bool syncByRevision;
+  bool usesRevisionsAsDocumentIds;
+  // Deprecated, and not documented anymore
+
+  std::string id;
+
+  // TODO: Maybe this is better off with a transformator Uint -> col_type_e
+  [[nodiscard]] TRI_col_type_e getType() const noexcept {
+    return TRI_col_type_e(type);
+  }
+};
+
+template<class Inspector>
+auto inspect(Inspector& f, PlanCollection& planCollection) {
+  return f.object(planCollection)
+      .fields(
+          f.field("name", planCollection.name)
+              .invariant([](auto const& n) -> inspection::Status {
+                if (n.empty()) {
+                  return {"Name cannot be empty."};
+                }
+                return inspection::Status::Success{};
+              }),
+          f.field("id", planCollection.id).fallback(""),
+          f.field("waitForSync", planCollection.waitForSync).fallback(false),
+          f.field("isSystem", planCollection.isSystem).fallback(false),
+          f.field("doCompact", planCollection.doCompact).fallback(false),
+          f.field("cacheEnabled", planCollection.cacheEnabled).fallback(false),
+          f.field("isVolatile", planCollection.isVolatile).fallback(false),
+          f.field("syncByRevision", planCollection.syncByRevision)
+              .fallback(true),
+          f.field("usesRevisionsAsDocumentIds",
+                  planCollection.usesRevisionsAsDocumentIds)
+              .fallback(true),
+          f.field("numberOfShards", planCollection.numberOfShards)
+              .fallback(1ULL)
+              .invariant([](auto const& value) -> inspection::Status {
```

Comment:
Numpty check; when this invariant fails, will the error message tell you which value needs to be >0?

Reply:
Yes it does:
```
http://localhost:8529/_api/collection -d '{"name": "test", "numberOfShards": -2}'
{"code":400,"error":true,"errorMessage":"Number out of range on path numberOfShards","errorNum":10} 
```

```
http://localhost:8529/_api/collection -d '{"name": "test", "numberOfShards": 0}'
{"code":400,"error":true,"errorMessage":"Value has to be > 0 on path numberOfShards","errorNum":10}
```